src/main/java/seedu/duke/ui/UI.java

```diff
@@ -1,4 +1,120 @@
 package seedu.duke.ui;
 
+import seedu.duke.task.Task;
+
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import static seedu.duke.common.Constants.NEWLINE;
+
 public class UI {
+    private final Scanner in;
+
+    public UI() {
+        in = new Scanner(System.in);
+    }
+
+    public String readCommand() {
+        return in.nextLine();
+    }
+
+    public boolean getIsTaskGraded() {
+        System.out.print("Is this task graded? (Y / N)" + NEWLINE);
+        String userInput = readCommand();
+        while (!userInput.equals("Y") && !userInput.equals("N")) {
+            System.out.print("Please enter \"Y\" or \"N\"" + NEWLINE);
+            userInput = readCommand();
+        }
+        return userInput.equals("Y");
+    }
+
+    public ArrayList<Task> getFilteredTasks(ArrayList<Task> taskList, boolean isDone) {
+        ArrayList<Task> tasksToModify = new ArrayList<>();
+        for (Task task : taskList) {
+            if (task.getDone() == isDone) {
+                tasksToModify.add(task);
+            }
+        }
+        return tasksToModify;
+    }
+
+    public ArrayList<Task> getChosenTasks(ArrayList<Task> taskList) {
+        // simulating the parser's checkIndex() method
+        ArrayList<Integer> indices = new ArrayList<>();
+        String input = readCommand();
+        String[] tokens = input.split(" ");
+        for (String token : tokens) {
+            indices.add(Integer.parseInt(token));
```

Comment:
Is there a check for validity of the token? (E.g. non-int input, negative value or out of bounds)